presto-main/src/main/java/com/facebook/presto/util/maps/IdentityLinkedHashMap.java

```diff
@@ -142,4 +133,116 @@ public int size()
             }
         };
     }
+
+    private class KeySet
```

Comment:
Well, then, unless we need all this code, I would prefer not adding it. It's just more stuff that can go wrong, that we have to maintain, etc.